cmd/cosign/cli/verify/verify.go

```diff
@@ -114,7 +115,7 @@ func (c *VerifyCommand) Exec(ctx context.Context, images []string) (err error) {
 		co.ClaimVerifier = cosign.SimpleClaimVerifier
 	}
 
-	if c.keylessVerification() {
+	if c.keylessVerification() || c.VerifyTlog {
```

Comment:
Oops, just realized you did add this in: This is currently implementing forced online I think, at least on the CLI layer. I would rename it to that, because if there's a bundle in the OCI registry, does that "count"?

Reply:
I'm not sure I understand the issue completely 🤔 Do you mean the keylessVerification check should actually be a forcedOnlineVerification check?

Reply:
I think it's confusing too... `c.VerifyTlog` is setting a RekorClient for ONLINE lookups. Not just any tlog verification (in case it comes from a bundle).

Reply:
Does it mean "any tlog verification, including from a bundle"?

Reply:
that's what i intended!

Reply:
My personal preference is allowing the client to specify additional online verification, rather than just any

When Rekor was sharded, we never detected that w/ probers because we never had a way to explicitly say go check Rekor online for a proof and all OCI images have a bundle. If we use this flag, we never get the path of supporting bundle verification and online inclusion proofs (which are stronger)

I don't think it'll force online -- if a bundle is supplied and works i think it exits early. but please check!

Reply:
TBH want to do the verify path in a follow-up? the sign path in this PR looks fine to me

Reply:
+1 to doing verify in a follow up

Reply:
sure, sign was my initial plan, but the e2e tests won't pass without verify. i can just keep those portions commented out for now.

Reply:
What if we construct a valid bundle for the test?

Reply:
i think it'll probably depend on what flags/behavior we implement in the verify PR, i think it makes more sense to update the e2e test in that one (right now the test is intended to confirm that signing/verifying with a managed key + online tlog lookup works)